src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -0,0 +1,626 @@
+    /// @param port service port
+    void mdns_service_add(const char *service, uint16_t port);
+
+    /// Remove service from mDNS server.
+    /// @param service service, e.g. _openlcb-can._tcp
+    void mdns_service_remove(const char *service);
+
+    /// Lookup an mDNS name.
+    /// @param service servicename to lookup
+    /// @param hints hints about limiting the types of services that will
+    ///        respond
+    /// @param addrinfo structure containing one or more service addressess that
+    ///        match the enquery, else nullptr if no matching service found
+    /// @return 0 upon success, or appropriate EAI_* error on failure, use
+    ///         ::freeaddrinfo() to free up memory allocated to the non nullptr
+    ///         *addr returned
+    int mdns_lookup(
+        const char *service, struct addrinfo *hints, struct addrinfo **addr);
+
+    /// Start continuous scan for mDNS service name.
+    /// @param service servicename to scan
+    void mdns_scan(const char *service);
+
+protected:
+    /// WiFi STA or AP credentials.
+    struct WiFiConfigCredentialsNVS
+    {
+        char ssid_[33]; ///< STA SSID
+        uint8_t sec_; ///< STA security mode
+        char pass_[64]; ///< STA password
+    };
+
+    /// C structure version of the WiFi configuration settings.
+    struct WiFiConfigNVS
+    {
+        uint32_t magic_; ///< magic number to detect initialization
+        WlanRole mode_; ///< role that the device will operate (AP, STA, etc.)
+        WiFiConfigCredentialsNVS ap_; ///< access point configuration
+        WiFiConfigCredentialsNVS sta_[7]; ///< station profiles
+    };
+
+    /// Private configuration metadata.
+    struct ConfigPrivate
+    {
+        WiFiConfigCredentialsNVS last_; ///< last STA successfully connected to
+        uint8_t channelLast_; ///< last channel successfully connected with
+    };
+
+    /// Magic number to detect initialization
+    static constexpr uint32_t WIFI_CONFIG_INIT_MAGIC = 0x6160CBC6;
+
+    /// Default STA SSID.
+    static constexpr char DEFAULT_STA_SSID[] = "LAYOUTWIFI";
+
+    /// NVS key for the WiFi user config.
+    static constexpr char NVS_KEY_USER_NAME[] = "user";
+
+    /// NVS key for the WiFi private config.
+    static constexpr char NVS_KEY_LAST_NAME[] = "last";
+
+    /// Maximum length of a stored SSID not including '/0' termination.
+    static constexpr size_t MAX_SSID_SIZE =
+        sizeof(WiFiConfigCredentialsNVS::ssid_) - 1;
+    static_assert(MAX_SSID_SIZE == 32, "Invalid maximum SSID length.");
+
+    /// Maximum length of a stored password not including '/0' termination.
+    static constexpr size_t MAX_PASS_SIZE =
+        sizeof(WiFiConfigCredentialsNVS::pass_) - 1;
+    static_assert(MAX_PASS_SIZE == 63, "Invalid maximum password length.");
+
+    /// Reset configuration to factory defaults.
+    void factory_default();
+
+    nvs_handle_t cfg_; ///< handle to the nvs config for wifi configuration
```

Comment:
From @balazsracz: it would be preferable to have all class variables next to each other (at the very bottom). Right now they are split between far away places in the source code.